File: frontend/transport-management-system-web/src/app/types/order.ts
import { ICustomer } from "./customer";

export interface IOrder {
  id: number;
  customerId: number;
  customerName: string;
  customerMatricule: string;
  customerCity?: string;
  reference: string;
  type: string;
  weight: number;
   weightUnit: string;
  status: OrderStatus;
  createdDate: Date;
   deliveryDate?: Date | string;
  deliveryAddress?: string;
  notes?: string;

  hasDelivery?: boolean;
  customer?: ICustomer;
  sourceSystem?: string;
  zoneId?: number;
  zoneName?: string;
}
export interface CreateOrderDto {
  customerId?: number;
  reference?: string;
  weight?: number;
  weightUnit: string;

  deliveryAddress?: string;
  notes?: string;
    customerCity?: string;
      deliveryDate?: string;
}

export interface UpdateOrderDto {
  customerId?: number;
  reference?: string;
  weight?: number;
  weightUnit: string;
  status?: OrderStatus;
  deliveryAddress?: string;
  notes?: string;

}

export enum OrderStatus {
  Pending = 'pending',
  ReadyToLoad = 'readyToLoad',
  InProgress = 'inProgress',
  Received = 'received',
  Closed = 'closed',
  Cancelled = 'cancelled'
}

export function getOrderStatusText(status: OrderStatus): string {
  switch (status) {
    case OrderStatus.Pending:
      return 'En attente';
    case OrderStatus.ReadyToLoad:
      return 'À charger';
    case OrderStatus.InProgress:
      return 'En cours de livraison';
    case OrderStatus.Received:
      return 'Réception';
    case OrderStatus.Closed:
      return 'Clôturée';
    case OrderStatus.Cancelled:
      return 'Annulée';
    default:
      return String(status);
  }
}



export function getOrderStatusClass(status: OrderStatus): string {
  switch (status) {
    case OrderStatus.Pending:
      return 'status-pending';
    case OrderStatus.ReadyToLoad:
      return 'status-ready';
    case OrderStatus.InProgress:
      return 'status-in-progress';
    case OrderStatus.Received:
      return 'status-received';
    case OrderStatus.Closed:
      return 'status-closed';
    case OrderStatus.Cancelled:
      return 'status-cancelled';
    default:
      return '';
  }


}

File: backend/TransportManagementSystem/Entity/Order.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace TransportManagementSystem.Entity;

public class Order
{
    [Key]
    [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
    public int Id { get; set; }

    [Required]
    public DataSource SourceSystem { get; set; } = DataSource.TMS;

    [StringLength(50)]
    public string? ExternalId { get; set; }

    [Required]
    public int CustomerId { get; set; }

    [ForeignKey("CustomerId")]
    public Customer Customer { get; set; }

    [Required]
    [StringLength(50)]
    public string Reference { get; set; } = string.Empty;

    public string? Type { get; set; } // Shampoo, Gel, Toothpaste, etc.

    [Required]
    [Column(TypeName = "decimal(10,2)")]
    public decimal Weight { get; set; } // in kg

    [Required]
    public OrderStatus Status { get; set; } = OrderStatus.Pending;

    [Required]
    public DateTime CreatedDate { get; set; } = DateTime.UtcNow;

    public DateTime? UpdatedDate { get; set; }

    public DateTime? DeliveryDate { get; set; }

    public string? DeliveryAddress { get; set; }

    [StringLength(500)]
    public string? Notes { get; set; }

    public int Priority { get; set; } = 5; // 1-10, 10 = highest priority

    // Navigation property for deliveries
    public ICollection<Delivery> Deliveries { get; set; } = new List<Delivery>();

    // Additional optional fields
    public string? Dimensions { get; set; } // e.g., "30x40x50 cm"
    public string? SpecialInstructions { get; set; }

    [Required]
    [StringLength(10)]
    public string WeightUnit { get; set; } = "palette"; // valeur par défaut
}

public enum OrderStatus
{
    Pending,              // En attente (commande créée)
    ReadyToLoad,          // Prête au chargement (avant ou après acceptation chauffeur)
    Planned,              // Planifié (par l'opérateur)
    Accepted,             // Accepté (par le chauffeur mobile)
    LoadingInProgress,    // En cours de chargement (après confirmation)
    DeliveryInProgress,   // En cours de livraison (en route)
    Receipt,              // Réception (Livrée - arrivée destination)
    Cancelled             // Annulée
}
